src/App.tsx

```diff
@@ -1,62 +1,174 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
 import classNames from 'classnames';
-import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import {
+  getComments,
+  getPosts,
+  getUsers,
+} from './api/services';
+import { Post } from './types/Post';
+import { PostsList } from './components/PostsList';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [loadingPost, setLoadingPost] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [errorPost, setErrorPost] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [loadComment, setLoadComment] = useState(false);
+  const [loadSideBar, setLoadSideBar] = useState(false);
+  const [errorComment, setErrorComment] = useState(false);
+
+  const loadUsers = async () => {
+    try {
+      const dataUsers = await getUsers();
+
+      setUsers(dataUsers);
+    } catch {
+      throw new Error('Error');
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  const loadPosts = async (userId: number) => {
+    setLoadingPost(true);
+    setErrorPost(false);
+
+    try {
+      const dataPosts = await getPosts(userId);
+
+      setPosts(dataPosts);
+    } catch {
+      setErrorPost(true);
+    } finally {
+      setLoadingPost(false);
+    }
+  };
+
+  const handleSelectUser = useCallback((user: User) => {
+    setLoadSideBar(false);
+    setSelectedUser(user);
+    loadPosts(user.id);
+  }, []);
+
+  const loadComments = async (postId: number) => {
+    setLoadComment(true);
+    setErrorComment(false);
+    try {
+      const dataComments = await getComments(postId);
+
+      setComments(dataComments);
+    } catch {
+      setErrorComment(true);
+    } finally {
+      setLoadComment(false);
+    }
+  };
+
+  const handleSelectPost = useCallback((post: Post | null) => {
+    setLoadSideBar(true);
+    setSelectedPost(post);
+
+    if (post) {
+      loadComments(post.id);
+    }
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+
+                <UserSelector
```

Comment:
```suggestion
                <UserSelector
```
Don't leave empty lines between parent and child elements